core/chain_makers.go

```diff
@@ -340,3 +366,98 @@ func (cr *fakeChainReader) GetGasUsedInChain(block *types.Block, length int) int
 func (cr *fakeChainReader) GetExternalBlocks(header *types.Header) ([]*types.ExternalBlock, error) {
 	return nil, nil
 }
+
+// GenerateNetwork creates a network of n blocks in c contexts. The first block's
```

Comment:
Just checking my understanding aligns with this description: If I'm reading the code correctly, this function is hard-coded to always generate zone, region, & prime blocks (so `c == 3` always), but only in one slice. Never in locations or chains outside of the given slice, correct?

Reply:
Correct. Current implementation of the Generator is one slice. A future iteration can expand to cover the entire main network.